Tactics/Reflect/ProgramInfo.lean

```diff
@@ -17,81 +17,318 @@ Furthermore, we define a persistent env extension to store `ProgramInfo` in.
 
 open Lean Meta Elab.Term
 
+initialize
+  registerTraceClass `ProgramInfo
+
+structure InstInfo where
+  /-- the raw instruction, as a bitvector -/
+  rawInst : BitVec 32
+
+  /-- the decoded instruction, as a normalized(!) `Expr` of type `ArmInst`.
+  That is, `decode_raw_inst <rawInst>` should be def-eq to `some <decodeInst>`.
+
+  NOTE: a `none` value indicates that this expression has not been computed yet,
+  *not* that `decode_raw_inst` returned `none`-- we assume that
+  all instructions for which we generate an `InstInfo` are well-formed -/
+  decodedInst? : Option Expr
```

Comment:
Consider making a different inductive for this?

```lean
inductive OnDemand (a : Type) where 
| NotYetDemanded
| Computed (v : a)
```

This way, the semantics are utterly clear.

Reply:
Added